beacon_node/network/src/beacon_processor/mod.rs

```diff
@@ -1336,6 +1336,17 @@ impl<T: BeaconChainTypes> BeaconProcessor<T> {
             &[work.str_id()],
         );
 
+        let notify_execution_layer = if self
+            .network_globals
+            .sync_state
+            .read()
+            .is_syncing_finalized()
+        {
+            NotifyExecutionLayer::No
+        } else {
+            NotifyExecutionLayer::Yes
+        };
+
```

Comment:
This is a minor point, but I think we might _only_ want to compute this for `ChainSegment`. That's more conservative (we'll always verify RPC and Gossip blocks) while still achieving our goal of speeding up sync. It also saves us locking the `sync_state` for every type of message processed, e.g. attestations, where we don't actually need it.

Reply:
Ah, that does seem better! I will make the changes.

Reply:
I was playing around with making the new type FinalizedSignatureVerifiedBlock like you had suggested, since then we could remove the notify_execution_layer parameter from a lot of the functions. Because all the fields of SignatureVerifiedBlock are private I was having trouble making the SignatureVerifiedBlock into a Finalized one. I could have added getter methods to SignatureVerifiedBlock, but I assumed that would be undesirable. Do you think I should make the new type since there is really only one path I saw in the code where notify_execution_layer: NotifyExecutionLayer is actually dependent on the sync_state?

Reply:
Why do the CI tests seem to fail non-deterministically? In the other PR I'm working on this same group of tests failed: test-suite / release-tests-windows (pull_request). Is there something I can do to ensure things pass?